camayoc/tests/qcs/test_scan.py

```diff
@@ -0,0 +1,185 @@
+:upstream: yes
+"""
+
+import logging
+import pytest
+import time
+
+from camayoc import config
+from camayoc.exceptions import ConfigFileNotFoundError
+from camayoc.qcs_models import (
+    HostCredential,
+    NetworkProfile,
+    Scan,
+)
+
+
+def profiles():
+    """Gather profiles from config file."""
+    try:
+        profs = config.get_config()['qcs']['profiles']
+    except ConfigFileNotFoundError:
+        profs = []
+    return profs
+
+
+LOGGER = logging.Logger(name='camayoc')
+
+
+def wait_until_complete(scan, timeout=120):
+    """Wait until the scan has either failed or completed.
+
+    This method should not be called on scan jobs that have not yet been
+    created, are paused, or are canceled.
+
+    The default timeout is set to 120 seconds, but can be overridden if it is
+    anticipated that a scan may take longer to complete.
+    """
+    if scan.status and scan.status() in ['paused', 'cancelled']:
+        LOGGER.critical(msg=('Camayoc Warning: Waiting for a job that will not'
+                             ' start because it is paused or canceled'))
+        return
+    while (
+        not scan.status() or not scan.status() in [
+            'failed',
+            'completed']) and timeout > 0:
+        time.sleep(5)
+        timeout -= 5
+
+
+@pytest.mark.parametrize('profile', profiles(), ids=[
+                         '{}-{}'.format(
+                             p['name'],
+                             p['auths']) for p in profiles()]
+                         )
+def test_create(shared_client, cleanup, profile, isolated_filesystem):
+    """Run a scan on a system and confirm it completes.
+
+    :id: ee3b0bc8-1489-443e-86bb-e2a2349e9c98
+    :description: Create a scan and assert that it completes
+    :steps:
+        1) Create host credential
+        2) Send POST with data to create network profile using the host
+           credential to the profile endpoint.
+        3) Create a scan
+        4) Assert that the scan completes and a fact id is generated
+    :expectedresults: A scan is run and has facts associated with it
+    """
+    cfg = config.get_config()
+
+    auth_ids = []
+    for auth in cfg['qcs'].get('auths'):
+        if auth['name'] in profile['auths']:
+            cred = HostCredential(
+                client=shared_client,
+                ssh_keyfile=auth['sshkeyfile'],
+                username=auth['username'],
+            )
+            cred.create()
+            cleanup.append(cred)
+            auth_ids.append(cred._id)
+
+    netprof = NetworkProfile(
+        client=shared_client,
+        hosts=profile['hosts'],
+        credential_ids=auth_ids,
+    )
+    netprof.create()
+    scan = Scan(profile_id=netprof._id)
+    scan.create()
+    wait_until_complete(scan)
+    assert scan.status() == 'completed'
+    assert isinstance(scan.read().json().get('fact_collection_id'), int)
+    # add the ids to the lists to destroy after the test is done
+    cleanup.append(netprof)
```

Comment:
I think netprof can be added just after its creation to the cleanup list since if any of the asserts above fail it won't get removed.